ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -137,7 +137,7 @@ public boolean addRemoteTransaction(final Transaction transaction) {
     return transactionAdded;
   }
 
-  boolean addTransactionHash(final Hash transactionHash) {
+  synchronized boolean addTransactionHash(final Hash transactionHash) {
```

Comment:
This also makes localTransactionHashesAdded synced when all we care about is `newPooledHashes`. Perhaps just `synchronized (newPooledHashes) { ... ` around the add and remove?

Reply:
sure thing.

Reply:
done